app/controllers/admin/users_controller.rb

```diff
@@ -150,7 +150,8 @@ def suspend
         suspend_reason: params[:reason],
         full_suspend_reason: user_history.try(:details),
         suspended_till: @user.suspended_till,
-        suspended_at: @user.suspended_at
+        suspended_at: @user.suspended_at,
+        suspended_by: BasicUserSerializer.new(current_user, root: false).as_json
```

Comment:
Shouldn't we be using the `@user.suspend_record.acting_user` instead? Otherwise `current_user` will always be the current (admin) user.

Reply:
In this context I think this is fine as the action is triggered by an admin user and this user (`current_user`) is returned in the response to be rendered.

Silence user action works exactly like this too

https://github.com/discourse/discourse/blob/c5c9b2eced7c3d9dae1a0ead432b5efade4eb1bd/app/controllers/admin/users_controller.rb#L365-L371

The full admin user details will return the `suspend_record` and correctly shows the acting user (tested this also locally that it works as expected)

Reply:
Right, my bad, I didn't look at the code of the whole function and `current_user` and `suspend_record.acting_user` are indeed the same user.